configs/xlml/tensorflow/solutionsTeam_tf_nightly_supported_config.py

```diff
@@ -182,10 +185,12 @@ def export_env_variable(is_pod: bool) -> str:
 def get_tpu_runtime(is_pod: bool) -> str:
   """Get TPU runtime image."""
   if is_pod:
-    return vm_resource.RuntimeVersion.VM_NIGHTLY_POD.value
-  return vm_resource.RuntimeVersion.VM_NIGHTLY.value
+    return vm_resource.RuntimeVersion.TPU_VM_TF_NIGHTLY_POD.value
+  return vm_resource.RuntimeVersion.TPU_VM_TF_NIGHTLY.value
 
 
 def create_tpu_name(test_name: str, tpu_version: str, tpu_cores: int) -> str:
-  """Create a custom TPU name."""
-  return f"{test_name}-v{tpu_version}-{tpu_cores}-{str(uuid.uuid4())}"
+  """Create a custom TPU name with hash value of the current date."""
+  date_suffix = date.today().strftime("%Y%m%d")
+  date_hash = hashlib.sha256(str(date_suffix).encode("utf-8")).hexdigest()[:36]
```

Comment:
Are you sure this works? I believe this part will run once when parsing the DAG, not each time the DAG is invoked.

Reply:
`I believe this part will run once when parsing the DAG` that was my understanding till Mark and I met TPU name consistency issue recently. Basically, there will be a different tpu_name when running the training process with `--tpu=tpu_name` (this is not affecting the SSH part, as we passed it as XCOM).

Reply:
I tried this code snippet

```
  tpu_name = create_tpu_name(test_name, tpu_version, tpu_cores)
  Variable.set(f"{test_name}_tpu_name", tpu_name)
  tpu_name_param = Variable.get(f"{test_name}_tpu_name") if is_pod else "local"

...

def create_tpu_name(test_name: str, tpu_version: str, tpu_cores: int) -> str:
  """Create a custom TPU name with hash value of the current date."""
#   date_suffix = date.today().strftime("%Y%m%d")
#   date_hash = hashlib.sha256(str(date_suffix).encode("utf-8")).hexdigest()[:36]
  return f"{test_name}-v{tpu_version}-{tpu_cores}-{str(uuid.uuid4())}"
```

Those are what I have observed:
* Variable is set up for ResNet v4-32: https://screenshot.googleplex.com/9oMW7xvCiMzKpYC
* TPU name is different: 
    * name from generate_tpu_name: tf_resnet_imagenet-v4-32-19b3586e-1b28-4686-91c5-ee99dedbf5b1
    * name from 1st run_model: tf_resnet_imagenet-v4-32-5d786247-dd9b-446b-8aaf-7e334c210ac4
    * name from 2nd run_model: tf_resnet_imagenet-v4-32-642ab819-4db8-4ff5-94b8-8f97cb4e8fcd

Reply:
I would expect the name to change once every time the DAG parser runs, which runs every minute or two. I'm actually more surprised if anything that the name stays consistent for a  whole run, the DAG can actually get rebuilt while tasks are in progress. But if the command gets encoded as an XCOM somewhere, then that will stay consistent.

Reply:
Yes, that's my current understanding. Like discussed offline, I will have a try with env variable in fabric run.

Reply:
I changed back to the original implementation for `generate_tpu_name()`, but adding `Variable.set()` to set TPU name as an airflow variable. In this case, helper functions can easily get that TPU name value, and this is not needed to be in a task or operator like XCOM. Tested in the Airflow (including retry), and updated the description. 

I keep the current implementation in ssh, and think it's fine. So we could get TPU names in 2 ways: 1) for TF tensorflow tests, get the name through airflow viable by passing the benchmark ID as the key; or 2) use passed xcom value directly between tasks. Let me know your thoughts!